### PR TITLE
Only test library load and port listing on CI

### PR DESCRIPTION
**Pull request by samuelpowell**

### CODE REVIEW

**Comment by samuelpowell** [comment]
@andrewadare changing permissions on `/dev/tty*` allows tests to run when operating in debug mode, but still no dice here. As such, I'm out of ideas! 

Of course, even when the tests did run on CI, they didn't do much. So further to our conversations regarding the utility of CI tests for this package, this PR simply calls `list_ports()` then exits when running on CI. Local tests are not affected. This will at least allow us to ensure the library is installing and can be successfully called.

Whilst doing this I accidentally pushed a commit to master. If you want to proceed with this PR I'll resolve the conflict, else I'll undo those changes. I'll put protection on the master branch to avoid trouble in the future!

Let me know what you think.

File: .travis.yml [hunk 1]
```diff
@@ -7,10 +7,12 @@ julia:
 #  - release
   - 0.7
   - 1.0
+  - 1.1
   - nightly
 ## uncomment the following lines to allow failures on nightly julia
 ## (tests will run but not make your overall status red)
-before_instal: sudo adduser $USER dialout
+before_instal:
```
Comment (andrewadare) [inline .travis.yml]: I believe this should be `before_install`.

**Comment by andrewadare** [comment]
Thanks @samuelpowell for working on this. Given the challenge of testing a library like this in a CI context, I think this is the best we can do. I support merging this PR. No problem that you pushed to master; should be simple to resolve. Feel free to merge it once you're satisfied.

I think once we have this, we can undertake to finally register this package.

**Comment by samuelpowell** [comment]
Thanks @andrewadare . The rather embarrassing typo in the travis file explains differences in debug mode! But in any case, exploring what the correct permissions enable is a task for another day.

**Pull request merged**